src/behaviors/Visibility/Visibility.js

```diff
@@ -171,108 +171,76 @@ export default class Visibility extends Component {
   }
 
   calculations = {
-    topPassed: false,
     bottomPassed: false,
-    topVisible: false,
     bottomVisible: false,
     fits: false,
     passing: false,
-    onScreen: false,
     offScreen: false,
+    onScreen: false,
+    topPassed: false,
+    topVisible: false,
   }
 
-  firedCallbacks = []
+  occurredCallbacks = []
+
+  // ----------------------------------------
+  // Lifecycle
+  // ----------------------------------------
 
   componentWillReceiveProps({ continuous, once }) {
-    const cleanOut = continuous !== this.props.continuous || once !== this.props.once
-    if (cleanOut) this.firedCallbacks = []
+    const cleanHappened = continuous !== this.props.continuous || once !== this.props.once
+
+    // Heads up! We should clean up array of happened callbacks, if values of these props are changed
+    if (cleanHappened) this.occurredCallbacks = []
   }
 
   componentDidMount() {
     if (!isBrowser) return
 
     const { context, fireOnMount } = this.props
 
-    context.addEventListener('scroll', this.handleScroll)
-    if (fireOnMount) this.handleUpdate()
+    context.addEventListener('scroll', this.handleUpdate)
```

Comment:
nit: suggest keeping `handleScroll` for the scroll handler.  It makes it clear when/why it is called.

Reply:
This comes from #2091, this handler will also handle `resize` event